lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -149,19 +149,58 @@ def cmd_workspace(*args)
       end
       framework.db.workspace = workspace
     elsif deleting and names
-      status_msg, error_msg = framework.db.delete_workspaces(names)
-      print_msgs(status_msg, error_msg)
+      ws_ids_to_delete = []
+      starting_ws = framework.db.workspace
+      names.each do |n|
+        ws_ids_to_delete << framework.db.find_workspace(n).id
+      end
+      deleted = framework.db.delete_workspaces(ids: ws_ids_to_delete)
+      process_deleted_workspaces(deleted, starting_ws)
     elsif delete_all
-      status_msg, error_msg = framework.db.delete_all_workspaces()
-      print_msgs(status_msg, error_msg)
+      ws_ids_to_delete = []
+      starting_ws = framework.db.workspace
+      framework.db.workspaces.each do |ws|
+        ws_ids_to_delete << ws.id
+      end
+      deleted = framework.db.delete_workspaces(ids: ws_ids_to_delete)
+      process_deleted_workspaces(deleted, starting_ws)
     elsif renaming
       if names.length != 2
         print_error("Wrong number of arguments to rename")
         return
       end
 
-      old, new = names
-      framework.db.rename_workspace(old, new)
+      ws_to_update = framework.db.find_workspace(names.first)
+      unless ws_to_update
+        print_error("Workspace '#{names.first}' does not exist")
+        return
+      end
+      opts = {
+          id: ws_to_update.id,
+          name: names.last
+      }
+      begin
+        if names.last == Msf::DBManager::Workspace::DEFAULT_WORKSPACE_NAME
+          print_error("Unable to rename a workspace to '#{Msf::DBManager::Workspace::DEFAULT_WORKSPACE_NAME}'")
+          return
+        end
+        updated_ws = framework.db.update_workspace(opts)
+        if updated_ws
+          framework.db.workspace = updated_ws if names.first == framework.db.workspace.name
+          print_status("Renamed workspace: #{updated_ws.name}")
```

Comment:
Minor: The message reads a little confusing, if I rename workspace test to new-workspace-name I would see the message "Renamed workspace: new-workspace-name". new-workspace-name is not the name of the workspace I renamed, but rather what I renamed another to. Consider changing to a more complete message, for example, "Renamed workspace #{old-name} to #{new-name}".